pkg/endpoint/bpf.go

```diff
@@ -345,15 +383,44 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir string) error {
 	owner.GetCompilationLock().RLock()
 	defer owner.GetCompilationLock().RUnlock()
 
-	e.Mutex.RLock()
+	e.Mutex.Lock()
+
+	// If endpoint was marked as disconnected then
+	// it won't be regenerated.
+	// When building the initial drop policy in waiting-for-identity state
+	// the state remains unchanged
+	if e.GetStateLocked() != StateWaitingForIdentity && !e.BuilderSetStateLocked(StateRegenerating) {
+		e.Mutex.Unlock()
+		log.WithFields(log.Fields{
+			logfields.EndpointID: e.StringID(),
+		}).Debugf("Skipping build due to invalid state: %s", e.state)
+		return fmt.Errorf("Skipping build due to invalid state: %s", e.state)
+	}
+
+	if e.Consumable != nil {
+		// Regenerate policy and apply any options resulting in the
+		// policy change.
+		if _, err = e.regeneratePolicy(owner, nil); err != nil {
+			e.Mutex.Unlock()
+			return fmt.Errorf("Unable to regenerate policy for '%s': %s",
```

Comment:
Unable -> unable (for consistency)

Reply:
Went the other way (consistently starting everything with a capital letter, dropping punctuation from the end).